[WIP] 1.13 Rendering patches

tterrag1098 opened the pull request:
* net.minecraft.client.renderer:
  - [x] ActiveRenderInfo
  - [x] BlockFluidRenderer
  - [x] BlockModelRenderer
  - [x] BlockModelShapes (Skipped)
  - [x] BlockRendererDispatcher
  - [x] BufferBuilder (Skipped, all fixes implemented in vanilla)
  - [x] ChestRenderer
  - [x] EntityRenderer (new patches for LightTexture and FogRenderer)
  - [x] InventoryEffectRenderer
  - [x] ItemModelMesher 
  - [x] ItemRenderer (now `FirstPersonRenderer`, add new patch for ItemMap)
  - [x] OpenGlHelper
  - [x] RenderGlobal
  - [x] RenderItem
  - [x] WorldVertexBufferUploader
* net.minecraft.client.renderer.block.model:
  - [x] BakedQuad
  - [x] BakedQuadRetextured
  - [x] FaceBakery
  - [x] IBakedModel
  - [x] ItemCameraTransforms
  - [x] ItemOverrideList
  - [x] ItemTransformVec3f
  - [ ] ModelBakery
  - [x] ModelBlockDefinition
  - [x] ModelManager
  - [x] ModelResourceLocation
  - [x] ModelRotation
  - [x] SimpleBakedModel
  - [x] Variant
  - [x] WeightedBakedModel
* net.minecraft.client.renderer.block.statemap:
  - [x] StateMap (gone)
* net.minecraft.client.renderer.chunk:
  - [x] ChunkRenderDispatcher
  - [x] RenderChunk
* net.minecraft.client.renderer.color:
  - [x] BlockColors
  - [x] ItemColors
* net.minecraft.client.renderer.entity:
  - [x] RenderEntityItem
  - [x] RenderFish
  - [x] RenderItemFrame
  - [ ] RenderLivingBase
  - [x] RenderManager
  - [x] RenderPlayer
  - [x] RenderVillager
  - [x] RenderZombieVillager
* net.minecraft.client.renderer.entity.layers:
  - [x] LayerArmorBase
  - [x] LayerBipedArmor
  - [x] LayerHeldItem
* net.minecraft.client.renderer.texture:
  - [x] Stitcher
  - [x] TextureAtlasSprite
  - [x] TextureManager
  - [x] TextureMap (This is a huge mess, only a first pass done)
  - [x] TextureUtil
* net.minecraft.client.renderer.tileentity:
  - [x] TileEntityItemStackRenderer (Deprecated, removed)
  - [x] TileEntityPistonRenderer (No longer necessary (?))
  - [x] TileEntityRendererDispatcher
  - [x] TileEntitySpecialRenderer
* net.minecraft.client.renderer.vertex:
  - [x] VertexFormat
  - [x] VertexFormatElement

tterrag1098 committed:
ActiveRenderInfo

tterrag1098 committed:
BlockFluidRenderer

tterrag1098 committed:
BlockModelRenderer

tterrag1098 committed:
BlockRendererDispatcher
BufferBuilder
ChestRenderer

tterrag1098 committed:
EntityRenderer, related patches and events

bs2609 commented on patches/minecraft/net/minecraft/client/renderer/LightTexture.java.patch:
```diff
@@ -0,0 +1,19 @@
++                  
++                  float[] colors = {f10, f11, f12};
++                  world.func_201675_m().getLightmapColors(p_205106_1_, f, f4, f5, colors);
++                  f10 = colors[0]; f11 = colors[1]; f10 = colors[2];
```
Minor typo here - second `f10` should be `f12`

tterrag1098 replied:
Darn, and I thought I caught them all.

tterrag1098 committed:
InventoryEffectRenderer

tterrag1098 committed:
ItemRenderer (now FirstPersonRenderer)
ItemModelMesher (and ItemMap tweak)

tterrag1098 committed:
Add forge comment in ItemMap patch

tterrag1098 committed:
Patch cleanup: typos and a missed patch

tterrag1098 committed:
Duplicate MC-4647 patch for new code path

tterrag1098 committed:
ItemRenderer
OpenGlHelper
RenderGlobal
WorldVertexBufferUploader
TileEntityRendererDispatcher
rendering TileEntity patches

tterrag1098 committed:
Merge remote-tracking branch 'upstream/1.13-pre' into render_patches

tterrag1098 committed:
Fix player camera viewport being inverted 180 degrees

tterrag1098 committed:
Fix diffuse lighting being doubly applied

tterrag1098 committed:
Fix bad temporary logic in selective resource reloads

tterrag1098 committed:
All of /renderer/block/model save ModelBakery

tterrag1098 committed:
ChunkRenderDispatcher & RenderChunk

tterrag1098 committed:
BlockColors
ItemColors
LayerArmorBase
LayerBipedArmor
LayerHeldItem

tterrag1098 committed:
/renderer/entity/

tterrag1098 committed:
/renderer/texture/

ichttt commented on patches/minecraft/net/minecraft/client/renderer/block/model/IBakedModel.java.patch:
```diff
@@ -1,6 +1,18 @@
+ import net.minecraft.util.EnumFacing;
+ import net.minecraftforge.api.distmarker.Dist;
+ import net.minecraftforge.api.distmarker.OnlyIn;
++import net.minecraftforge.client.extensions.IForgeBakedModel;
```
Import in patch

ichttt commented on patches/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java.patch:
```diff
@@ -0,0 +1,16 @@
++   public ChunkRenderDispatcher(int countRenderBuilders) {
+       int i = Math.max(1, (int)((double)Runtime.getRuntime().maxMemory() * 0.3D) / 10485760);
+       int j = Math.max(1, MathHelper.func_76125_a(Runtime.getRuntime().availableProcessors(), 1, i / 5));
++      if(countRenderBuilders < 0) countRenderBuilders = MathHelper.func_76125_a(j * 10, 1, i);
```
This parameter seems like it is never read from

tterrag1098 replied:
Intended, this is just a hook for mods

ichttt replied:
But it is not even written to a field in this patch, it just calculates something and then never accesses it. In 1.12 the result is written to the field.

tterrag1098 replied:
You're right, good catch.

ichttt commented on patches/minecraft/net/minecraft/client/renderer/entity/layers/LayerArmorBase.java.patch:
```diff
@@ -0,0 +1,98 @@
+ import com.google.common.collect.Maps;
+ import java.util.Map;
++
++import javax.annotation.Nullable;
```
Import

ichttt commented on patches/minecraft/net/minecraft/client/renderer/texture/TextureMap.java.patch:
```diff
@@ -0,0 +1,186 @@
+ import java.io.IOException;
+ import java.util.List;
+ import java.util.Map;
++import java.util.OptionalInt;
```
Import

ichttt reviewed (commented):
Just looked through the patches on github real quick, this is what I noticed. Not a complete review.

tterrag1098 committed:
Fix typo in ChunkRenderDispatcher and remove a few stray imports

tterrag1098 committed:
/renderer/tileentity - no patches, only removed dead code

tterrag1098 committed:
VertexFormatElement - already done!

tterrag1098 committed:
Fix models being baked to BLOCK instead of ITEM

tterrag1098 committed:
Fix botched TAS patch skipping uploading some sprites

tterrag1098 committed:
Move nametag render distance to an attribute
